environments/dev/main.tf

```diff
@@ -0,0 +1,35 @@
+module "notification" {
+  source       = "../../modules/notification"
+  sender_email = var.sender_email
+}
+
+module "security" {
+  source = "../../modules/security"
+}
+
+module "pipeline" {
+  source                        = "../../modules/pipeline"
+  lambda_role_arn               = module.security.lambda_role_arn
+  sender_email                  = var.sender_email
+  serverless_sfn_role_arn       = module.security.serverless_sfn_role_arn
+  cloudwatch_serverless_sfn_arn = module.monitoring.cloudwatch_serverless_sfn_arn
+
+  depends_on = [module.security,
```

Comment:
this is an explicit dependency that you may not need (only in very specific use cases), since you are implicitly referring to the modules in the previous lines